tools/autograd/gen_variable_type.py

```diff
@@ -120,47 +399,128 @@ def unpack_args(option):
         option['unpacked_args'] = unpacked_args
```

Comment:
Again, can we isolate all modifications to option to be in a single place, and use maps local to each function if they're not shared in any way?